packages/devtools_app/lib/src/memory/memory_screen.dart

```diff
@@ -222,6 +226,28 @@ class MemoryBodyState extends State<MemoryBody>
     );
   }
 
+  void _refreshCharts() {
+    // Remove history of all plotted data in all charts.
+    eventChartController?.reset();
+    vmChartController?.reset();
+    androidChartController?.reset();
+
+    _recomputeChartData();
+  }
+
+  /// Recompute (attach data to the chart) for either live or offline data source.
+  void _recomputeChartData() {
+    if (eventChartController.traces.isNotEmpty) {
+      eventChartController.setupData();
+    }
+    if (vmChartController.traces.isNotEmpty) {
```

Comment:
nit: are these checks for empty data really needed? I would expect setupData should be trivial if there is no data.

Reply:
Only test needed is for Android Memory which isn't realized until the user clicks on the Android Memory button.  It should be in the Android's memory setupData only.  Removed these tests.